src/index.ts

```diff
@@ -303,6 +304,30 @@ async function appBoot() {
 		globalShortcut.unregisterAll();
 	} );
 
+	app.on( 'before-quit', ( event ) => {
+		if ( ! ipcHandlers.hasActivePushPullOperations() ) {
+			return;
+		}
+
+		const QUIT_APP_BUTTON_INDEX = 0;
+		const CANCEL_BUTTON_INDEX = 1;
+
+		const clickedButtonIndex = dialog.showMessageBoxSync( {
+			message: __( 'Sync in progress' ),
+			detail: __(
+				'There’s a sync operation in progress. Quitting the app will abort that operation. Are you sure?'
+			),
```

Comment:
I can't help myself, but somehow feel the following question would be more complete:

```suggestion
			detail: __(
				'There’s a sync operation in progress. Quitting the app will abort that operation. Are you sure you want to quit?'
			),
```

However, I am also good with keeping the copy as it is.

Reply:
That's fair 👍 I previously tried to keep this text short because it started out as the value for `message` (until I learned about `detail`). Nothing pushing us to keep it super short now, and I agree your wording makes more sense.

Reply:
Thanks for the update, Fredrik.